tests/Doctrine/Tests/ORM/Functional/UUIDGeneratorTest.php

```diff
@@ -20,18 +23,28 @@ class UUIDGeneratorTest extends OrmFunctionalTestCase
 {
     use VerifyDeprecations;
 
+    /**
+     * @group dbal2
+     */
     public function testItIsDeprecated(): void
     {
         $this->expectDeprecationWithIdentifier('https://github.com/doctrine/orm/issues/7312');
         $this->_em->getClassMetadata(UUIDEntity::class);
     }
 
+    /**
+     * @group dbal2
+     */
     public function testGenerateUUID(): void
     {
         if ($this->_em->getConnection()->getDatabasePlatform()->getName() !== 'mysql') {
             self::markTestSkipped('Currently restricted to MySQL platform.');
         }
 
+        if (! method_exists(Connection::class, 'getGuidExpression')) {
```

Comment:
I don't think `Connection` is the right class for this, I think you meant `AbstractPlatform`.

Reply:
Yup, that was it!